frontend/src/stimulus/controllers/dynamic/storages/automatically-managed-project-folders-form.controller.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * -- copyright
+ * OpenProject is an open source project management software.
+ * Copyright (C) 2023 the OpenProject GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 3.
+ *
+ * OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+ * Copyright (C) 2006-2013 Jean-Philippe Lang
+ * Copyright (C) 2010-2013 the ChiliProject Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * See COPYRIGHT and LICENSE files for more details.
+ * ++
+ */
+
+import { Controller } from '@hotwired/stimulus';
+
+export default class AutomaticallyManagedProjectFoldersFormController extends Controller {
+  static targets = [
+    'automaticallyManagedSwitch',
+    'applicationPasswordInput',
+  ];
+
+  static values = {
+    isAutomaticallyManaged: Boolean,
+  };
+
+  declare readonly automaticallyManagedSwitchTarget:HTMLElement;
+  declare readonly applicationPasswordInputTarget:HTMLElement;
+  declare readonly hasApplicationPasswordInputTarget:boolean;
+
+  declare isAutomaticallyManagedValue:boolean;
+
+  spotSwitchComponent!:HTMLElement;
+  boundUpdateDisplay!:(evt:InputEvent) => (void | boolean);
```

Comment:
I'd like to suggest some structural changes in the stimulus controller, that hopefully simplifies the code here:

- change signature `updateDisplay` to 

```typescript
public updateDisplay(_evt:InputEvent):void
```

- use it directly via an stimulus action on the spot switch component (if possible, needs testing)

if you want to check this together, I would gladly have a look at it. We also can do this in a follow up step.

But overall I think, this controller can be reduced by more then half.